### ISSUE TITLE
Questions and Issues with run_deconvolution and Code Implementation

### ISSUE THREAD

**Issue by liftvipi** [wwYinYin/SpatialcoGCN#3]
Hi

I have been recently researching spatial deconvolution and found your paper and code to be extremely helpful. However, I encountered some issues when attempting to use the run_deconvolution function with dataset7 instead of dataset11. The results I obtained were significantly different from those reported in your paper, and the performance on other simulated datasets was also not as expected. I would like to know if there are any additional post-processing steps required after running run_deconvolution.

Additionally, I have a few questions regarding specific parts of the code that I do not fully understand:

1.adata_to_cluster_expression Function:
Why is the expression level for each cell type summed (X_new[index] = sc_adata[sc_adata.obs['celltype'] == l].X.sum(axis=0)) rather than using the median or mean?

2.loss_function Function:
During training, the loss value eventually stabilizes at 2.0 and does not change. Could you provide insights into why this might be happening?

3.coGCN_training Function:
In each epoch, the predicted matrix pred_M is normalized by column (gene expression) using M_probs = pred_M / pred_M.sum(axis=0). Why is column normalization used instead of row (cell expression) normalization?

4.SingleCellDataset Class:
Loading data using the dataloader is quite slow, taking around 20 seconds. Is there any way to optimize this process?
I highly appreciate the method you have developed and would greatly benefit from your guidance on these issues. If possible, it would be even more helpful if you could provide the source code used in your experiments.

Thank you very much for your time and assistance.

The results obtained from run_deconvolution：
<img width="502" alt="屏幕截图 2024-11-26 183747" src="https://github.com/user-attachments/assets/807a103f-6929-4e9b-8e3c-f37475dc3459">
The results provided in the data you supplied：
<img width="734" alt="屏幕截图 2024-11-26 183816" src="https://github.com/user-attachments/assets/f5b9dd6a-5550-48c5-ab23-9dd18a07fec7">
The results obtained from run_deconvolution：
<img width="191" alt="屏幕截图 2024-11-26 183848" src="https://github.com/user-attachments/assets/874900f6-4d45-436d-85d8-eec9b636b3e5">
The results provided in the data you supplied：
<img width="278" alt="屏幕截图 2024-11-26 183924" src="https://github.com/user-attachments/assets/2b62771d-895e-45bb-b3b2-b32c1ec20892">

**Comment by wwYinYin** [comment]
Strictly following the instructions on GitHub to configure the environment may solve your problem

**Comment by wwYinYin** [comment]
The loss function may be one of the reason, we have changed the code in https://github.com/wwYinYin/SpatialcoGCN/blob/main/model/losses.py#L46

**Comment by ThesecondtoNone** [comment]
I have a question regarding the dataset `data/preudo_STData/dataset12(14)` mentioned in the author's tutorial within `simulated_data.py`. Where did you find it?thankyou！

**Comment by ThesecondtoNone** [comment]
May I ask what the missing dataset here is?Could you please reply to me?thankyou！

**Comment by wwYinYin** [comment]
data/preudo_STData/dataset12(14) This data has no practical significance. It is just a small data to debug whether there is any problem with my open source code.

**Comment by ThesecondtoNone** [comment]
> data/preudo_STData/dataset12(14) This data has no practical significance. It is just a small data to debug whether there is any problem with my open source code.

thankyou！

**Comment by ThesecondtoNone** [comment]
While I am running the run_impute.ipynb, I encountered some data content, such as data/DataUpload/Dataset2(9) and:
Spatial_file = DataDir + 'Insitu_count.txt'
location_file = DataDir + 'Locations.txt'
RNA_data_adata = sc.read_h5ad(DataDir + 'starmap_sc_rna.h5ad')，Where can I download these data? Could you please provide detailed addresses or data resources? Thank you!
